fix(schedule): use and logic to handle dow+dom (#32903)
renovatebot/renovate: docs/usage/configuration-options.md

This would mean that Renovate can run for 7 hours each night, plus all the time on weekends.
Note how the above example makes use of the "OR" logic of combining multiple schedules in the array.

<!-- prettier-ignore -->
!!! note

>     When both day of week and day of month are restricted in the schedule.
>     The behavior is to run when both the day of the month and day of week match
>     Example:
>     `* * 1-7 * 4` this schedule will be triggered on a Thursday which falls in the first week of the month.

Reviewer: ```suggestion
    If both the day of the week _and_ the day of the month are restricted in the schedule, then Renovate only runs when both the day of the month _and_ day of the week match!
    For example: `* * 1-7 * 4` means Renovate only runs on the _first_ Thursday of the month.
```

How about this?
